console/console-init/ui/src/modules/device/components/DeviceFilter/DeviceFilter.tsx

```diff
@@ -203,24 +233,24 @@ const DeviceFilter: React.FunctionComponent<IDeviceFilterProps> = (
   };
 
   const updateCriteria = (criteria: IDeviceFilterCriteria) => {
-    const filterObj = { ...filter };
+    const filterObj = JSON.parse(JSON.stringify(filter));
     const list = filterObj.filterCriteria;
     const index = findIndexByProperty(list, "key", criteria.key);
     list[index] = criteria;
     filterObj.filterCriteria = list;
     setFilter(filterObj);
   };
   const onStartTimeChange = (value: string) => {
```

Comment:
i would like to suggest create single function/event for handle setTime value. It will easy to manage the code and changes.
onStartTimeChange,onStartTimeFormatChange, onEndTimeChange and onEndTimeFormatChange.
these all function using for same purpose.